src/OAuth/OAuth2/Service/Reddit.php

```diff
@@ -51,6 +51,39 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
+    public function getAuthorizationUri(array $additionalParameters = array())
+    {
+        $parameters = array_merge(
+            $additionalParameters,
+            array(
+                'type'          => 'web_server',
+                'client_id'     => $this->credentials->getConsumerId(),
+                'redirect_uri'  => $this->credentials->getCallbackUrl(),
+                'response_type' => 'code',
+            )
+        );
+
+        $parameters['scope'] = implode(',', $this->scopes);
```

Comment:
There is a `getScopesDelimiter()` you can use for this:

https://github.com/Lusitanian/PHPoAuthLib/blob/master/src/OAuth/OAuth2/Service/AbstractService.php#L360